dl1_data_handler/image_mapper.py

```diff
@@ -1,744 +1,1036 @@
-                "rebinning",
-                "nearest_interpolation",
-                "bilinear_interpolation",
-                "bicubic_interpolation",
-                "image_shifting",
-                "axial_addressing",
-                "indexed_conv",
-            ]:
-                raise ValueError(
-                    "Hex conversion algorithm {} is not implemented.".format(map_method)
-                )
-            elif map_method in [
-                "image_shifting",
-                "axial_addressing",
-                "indexed_conv",
-            ] and camtype in ["ASTRICam", "CHEC", "SCTCam"]:
-                raise ValueError(
-                    "{} (hexagonal convolution) is not available for square pixel cameras.".format(
-                        map_method
-                    )
+        Returns
+        -------
+        numpy.ndarray
+            A numpy array of shape [img_width, img_length, N_channels].
+        """
+        # Reshape each channel and stack the result
+        images = np.concatenate(
+            [
+                (raw_vector[:, channel].T @ self.mapping_table).reshape(
+                    self.image_shape, self.image_shape, 1
                 )
+                for channel in range(raw_vector.shape[1])
+            ],
+            axis=-1,
+        )
+        return images
 
-            if map_method in [
-                "rebinning",
-                "nearest_interpolation",
-                "bilinear_interpolation",
-                "bicubic_interpolation",
-            ]:
-                self.image_shapes[camtype] = self.interpolation_image_shape[camtype]
-
-            # At the edges of the cameras the mapping methods run into issues.
-            # Therefore, we are using a default padding to ensure that the camera pixels aren't affected.
-            # The default padding is removed after the conversion is finished.
-            if map_method in ["image_shifting", "axial_addressing", "indexed_conv"]:
-                self.default_pad = 0
-            elif map_method == "bicubic_interpolation":
-                self.default_pad = 3
-            else:
-                self.default_pad = 2
-
-            if map_method != "oversampling" or camtype in [
-                "ASTRICam",
-                "CHEC",
-                "SCTCam",
-            ]:
-                self.image_shapes[camtype] = (
-                    self.image_shapes[camtype][0] + self.default_pad * 2,
-                    self.image_shapes[camtype][1] + self.default_pad * 2,
-                    self.image_shapes[camtype][2],
-                )
-            else:
-                self.image_shapes[camtype] = (
-                    self.image_shapes[camtype][0] + self.default_pad * 4,
-                    self.image_shapes[camtype][1] + self.default_pad * 4,
-                    self.image_shapes[camtype][2],
-                )
+    def _get_virtual_pixels(self, x_ticks, y_ticks, pix_x, pix_y):
```

Comment:
General suggestion: Add parameters meainig in "Doc Strings". You can use tools to generate it automatically in order to speed up this tedious task. Nobody like documentation 😂

Reply:
I did not add the parameters for the internal functions of the `ImageMapper` class, since I thought it was not necessary. Only for the one (`map_image()`) that can be called  from outside. If you are insisting, I can add them here as well ;-)